### PR TITLE
pdbx get_sequence return dict of seq

### PR DESCRIPTION
**Pull request by tjmier**
Accidentally deleted my last pull request on this topic. 

**Purpose:**
Modified biotite.structure.io.pdbx.get_sequence to return a dictionary of sequences instead of a list. This change provides a more intuitive way to access sequences by their chain identifiers.

**Details:**
Old Behavior: Returned a list of sequences.
New Behavior: Returns a dictionary where keys are derived from entity_poly.pdbx_strand_id (often equivalent to chain_id and atom_site.auth_asym_id in most cases). Dictionary values are the corresponding sequences.

**Example Usage:**
```
seq_dict = get_sequences(my_cif)
chain_A_seq = seq_dict["A"]
```
**Rationale:**
The change in return type improves the usability of the function by allowing direct access to sequences using chain identifiers, which is a common use case.

**Impact:**
This change is non-backwards compatible. Code that relies on the previous list-based return type will need to be updated.
Tests and examples have been updated to reflect this change in return type.

### CODE REVIEW

**Comment by padix-key** [comment]
For reference: This is the continuation of #600

**Review by padix-key — APPROVED** [review]

**Comment by padix-key** [comment]
Looks good to me. I will keep this PR open until patch version 0.41.2 was released (due to #605), which should arrive in the upcoming days.

File: src/biotite/structure/io/pdbx/convert.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def _filter(category, index):
 
 
 def get_sequence(pdbx_file, data_block=None):
-    """
+    """""
```
Comment (padix-key) [inline src/biotite/structure/io/pdbx/convert.py:114]: I just wanted to merge, then I saw this typo:
```suggestion
    """
```

**Comment by tjmier** [comment]
Excellent! Thank you for your patience. I learned a lot despite this being such a small change.

**Pull request merged**